File: symmetric-tree/symmetric-tree.py
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:

        def helper(node1, node2):
            if not node1:
                if not node2:
                    return True
                else:
                    return False
            if not node2:
                if not node1:
                    return True
                else:
                    return False
            return node1.val == node2.val and helper(node1.left, node2.right) and helper(node1.right, node2.left)
        
        return helper(root.left, root.right)

File: symmetric-tree/symmetric-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
bool helper(TreeNode* left, TreeNode* right) {
            if (left == nullptr || right == nullptr) return left == right;

            return (left -> val == right -> val) && helper(left -> right, right -> left) && helper(left -> left, right -> right); 
        }
    bool isSymmetric(TreeNode* root) {
        
        return helper(root -> left, root -> right);
    }
};
